diff -ur a/mlan/mlan_11n_rxreorder.c b/mlan/mlan_11n_rxreorder.c
--- a/mlan/mlan_11n_rxreorder.c	2024-10-01 09:55:31.723403237 +0000
+++ b/mlan/mlan_11n_rxreorder.c	2024-10-01 09:59:50.156409645 +0000
@@ -661,8 +661,8 @@
 	DBG_HEXDUMP(MCMD_D, "addba req", (t_u8 *)addba,
 		    sizeof(HostCmd_DS_11N_ADDBA_REQ));
 	if (priv->adapter->scan_processing) {
-		PRINTM(MERROR,
-		       "Scan in progress, ignore ADDBA Request event\n");
+//		PRINTM(MERROR,
+//		       "Scan in progress, ignore ADDBA Request event\n");
 		LEAVE();
 		return ret;
 	}
