From d429ea5dc24abbca634a6e76d0bdf89845742749 Mon Sep 17 00:00:00 2001
From: DD Fwdport <fwdport@dynamicdevices.co.uk>
Date: Sun, 19 Jul 2026 19:02:12 +0100
Subject: [PATCH 24/27] ASoC: tac5x1x: wire into codecs Makefile (6.12 -y
 style)

---
 sound/soc/codecs/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/codecs/Makefile b/sound/soc/codecs/Makefile
index 69cb0b39f220..3af5ad87c203 100644
--- a/sound/soc/codecs/Makefile
+++ b/sound/soc/codecs/Makefile
@@ -287,6 +287,7 @@ snd-soc-sta350-y := sta350.o
 snd-soc-sta529-y := sta529.o
 snd-soc-stac9766-y := stac9766.o
 snd-soc-sti-sas-y := sti-sas.o
+snd-soc-tac5x1x-y := tac5x1x.o
 snd-soc-tas5086-y := tas5086.o
 snd-soc-tas571x-y := tas571x.o
 snd-soc-tas5720-y := tas5720.o
@@ -687,6 +688,7 @@ obj-$(CONFIG_SND_SOC_STA350)   += snd-soc-sta350.o
 obj-$(CONFIG_SND_SOC_STA529)   += snd-soc-sta529.o
 obj-$(CONFIG_SND_SOC_STAC9766)	+= snd-soc-stac9766.o
 obj-$(CONFIG_SND_SOC_STI_SAS)	+= snd-soc-sti-sas.o
+obj-$(CONFIG_SND_SOC_TAC5X1X)	+= snd-soc-tac5x1x.o
 obj-$(CONFIG_SND_SOC_TAS2552)	+= snd-soc-tas2552.o
 obj-$(CONFIG_SND_SOC_TAS2562)	+= snd-soc-tas2562.o
 obj-$(CONFIG_SND_SOC_TAS2764)	+= snd-soc-tas2764.o
-- 
2.43.0

