diff --git a/board/freescale/imx93_11x11_evk/imx93_11x11_evk.c b/board/freescale/imx93_11x11_evk/imx93_11x11_evk.c
index 1b183b25287..5104a1d7cc9 100644
--- a/board/freescale/imx93_11x11_evk/imx93_11x11_evk.c
+++ b/board/freescale/imx93_11x11_evk/imx93_11x11_evk.c
@@ -330,7 +330,7 @@ static void set_fdt_file(void)
 	is_bd71837 = !ret && ((is_bd71837 & 0xf0) == 0xa0);
 
 	if (!is_bd71837)
-		env_set("fdt_file", "imx93-11x11-evk.dtb");
+		env_set("fdt_file", "imx93-jaguar-eink.dtb");
 	else
 		env_set("fdt_file", "imx93-11x11-evk.dtb");
 }
