diff --git a/board/freescale/imx93_evk/imx93_evk.c b/board/freescale/imx93_evk/imx93_evk.c
index 1b183b25287..5104a1d7cc9 100644
--- a/board/freescale/imx93_evk/imx93_evk.c
+++ b/board/freescale/imx93_evk/imx93_evk.c
@@ -332,6 +332,7 @@ int board_late_init(void)
 	env_set("board_name", "11X11_EVK");
 	env_set("board_rev", "iMX93");
 #endif
+	env_set("fdt_file", "imx93-jaguar-eink.dtb");
 	return 0;
 }
