diff --git a/drivers/net/wireless/microchip/wilc1000/cfg80211.c b/drivers/net/wireless/microchip/wilc1000/cfg80211.c
index b545d93c6e37..6bc6499aca9a 100644
--- a/drivers/net/wireless/microchip/wilc1000/cfg80211.c
+++ b/drivers/net/wireless/microchip/wilc1000/cfg80211.c
@@ -377,7 +377,7 @@ static int connect(struct wiphy *wiphy, struct net_device *dev,
 	}
 
 	if (wfi_drv->usr_scan_req.scan_result) {
-		netdev_err(vif->ndev, "%s: Scan in progress\n", __func__);
+#		netdev_err(vif->ndev, "%s: Scan in progress\n", __func__);
 		ret = -EBUSY;
 		goto out_error;
 	}
