diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index e02ef31da68e..8f4b84c4a0e9 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -709,6 +709,9 @@ static const struct xhci_driver_data reneses_data = {
 
 /* PCI driver selection metadata; PCI hotplugging uses this */
 static const struct pci_device_id pci_ids[] = {
+	{ PCI_DEVICE(0x16C3, 0xABCD),
+		.driver_data =  (unsigned long)&reneses_data,
+	},
 	{ PCI_DEVICE(0x1912, 0x0014),
 		.driver_data =  (unsigned long)&reneses_data,
 	},
