diff --git a/board/freescale/imx8mm_evk/imx8mm_evk.c b/board/freescale/imx8mm_evk/imx8mm_evk.c
index f27da09c655..62c0baf74bf 100644
--- a/board/freescale/imx8mm_evk/imx8mm_evk.c
+++ b/board/freescale/imx8mm_evk/imx8mm_evk.c
@@ -330,7 +330,7 @@ static void set_fdt_file(void)
 	is_bd71837 = !ret && ((is_bd71837 & 0xf0) == 0xa0);
 
 	if (!is_bd71837)
-		env_set("fdt_file", "imx8mm-evk.dtb");
+		env_set("fdt_file", "imx8mm-jaguar-sentai.dtb");
 	else
 		env_set("fdt_file", "imx8mm-evk-qca-wifi.dtb");
 }
