diff --git a/drivers/gpu/drm/panel/panel-sitronix-st7701.c b/drivers/gpu/drm/panel/panel-sitronix-st7701.c
index 7e12cbb48585..fcf45f60e67c 100644
--- a/drivers/gpu/drm/panel/panel-sitronix-st7701.c
+++ b/drivers/gpu/drm/panel/panel-sitronix-st7701.c
@@ -629,6 +629,10 @@ static int sit_panel_get_modes(struct drm_panel *panel, struct drm_connector *co
 	connector->display_info.height_mm = sit->height_mm;
 	mode->type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED;
 
+	// AJL - Fix up as the LCD display is smaller than the LCD controller
+	mode->hdisplay = 420;
+	mode->vdisplay = 640;
+
 	if (sit->vm.flags & DISPLAY_FLAGS_DE_HIGH)
 		*bus_flags |= DRM_BUS_FLAG_DE_HIGH;
 	if (sit->vm.flags & DISPLAY_FLAGS_DE_LOW)
