diff -ur kernel-source.org/sound/core/pcm_lib.c kernel-source/sound/core/pcm_lib.c
--- kernel-source.org/sound/core/pcm_lib.c	2024-07-30 11:04:15.940826525 +0000
+++ kernel-source/sound/core/pcm_lib.c	2024-07-30 11:08:10.058023485 +0000
@@ -2185,6 +2185,8 @@
 	if (err < 0)
 		return err;
 
+	pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 1\n");
+
 	is_playback = substream->stream == SNDRV_PCM_STREAM_PLAYBACK;
 	if (interleaved) {
 		if (runtime->access != SNDRV_PCM_ACCESS_RW_INTERLEAVED &&
@@ -2197,6 +2199,8 @@
 		writer = noninterleaved_copy;
 	}
 
+	pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 2\n");
+
 	if (!data) {
 		if (is_playback)
 			transfer = fill_silence;
@@ -2221,6 +2225,8 @@
 
 	nonblock = !!(substream->f_flags & O_NONBLOCK);
 
+	pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 3\n");
+
 	snd_pcm_stream_lock_irq(substream);
 	err = pcm_accessible_state(runtime);
 	if (err < 0)
@@ -2244,6 +2250,8 @@
 
 	avail = snd_pcm_avail(substream);
 
+	pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 4\n");
+
 	while (size > 0) {
 		snd_pcm_uframes_t frames, appl_ptr, appl_ofs;
 		snd_pcm_uframes_t cont;
@@ -2279,6 +2287,9 @@
 			err = -EBUSY;
 			goto _end_unlock;
 		}
+
+		pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 5\n");
+
 		snd_pcm_stream_unlock_irq(substream);
 		if (!is_playback)
 			snd_pcm_dma_buffer_sync(substream, SNDRV_DMA_SYNC_CPU);
@@ -2300,6 +2311,8 @@
 		if (err < 0)
 			goto _end_unlock;
 
+		pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 6\n");
+
 		offset += frames;
 		size -= frames;
 		xfer += frames;
@@ -2312,6 +2325,9 @@
 				goto _end_unlock;
 		}
 	}
+
+	pcm_warn(substream->pcm, "__snd_pcm_lib_xfer 7\n");
+
  _end_unlock:
 	runtime->twake = 0;
 	if (xfer > 0 && err >= 0)
